flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/history/HistoryServerTest.java

```diff
@@ -131,11 +132,11 @@ public void testHistoryServerIntegration() throws Exception {
 			archives = jmDirectory.listFiles();
```

Comment:
Is this while loop still needed after changing the verification approach or `HistoryServer` will not monitor job files created after its start?

Also a side note, would it be a more incapsulated approach to poll `MultipleJobsDetails` similar way in this loop until the expected, non-failed state reached instead of instrumenting `HistoryServer` internals for testing?

Reply:
Regarding the loop, I suppose we don't technically need it but it can speed up the test.

We had the polling approach in the past but it wasn't particularly stable and way more complicated than the current approach since you have to deal with quite a few error cases.

Reply:
How does the loop speed up the test? because it even introduces a sleep while `numExpectedArchivedJobs` just immediately gives a callback when everything is done. Does `numExpectedArchivedJobs` not basically subsume the loop time-wise? i.e. `numExpectedArchivedJobs` is done only when the loop is done

Regarding the polling approach I'm fine with it if it has been already discussed before.
It would be a separate issue anyways.

Reply:
With the current settings it can speed things up since the first poll of the HS is more likely to fetch all archives.

Another benefit is that it is easier to tell whether the archiving has failed or not. Without the first loop, if the archiving on the JM fails the HS we'd never execute the callback; in this case we'd only know that _something_ went wrong. With the dedicated loop to check the JM archiving it is easier to pinpoint which component is experiencing problems.